### PR TITLE
feat(host): limits host properties to renames

### PR DESCRIPTION
**Pull request by vsavkin**

### CODE REVIEW

File: modules/angular2/src/forms/directives/checkbox_value_accessor.ts [hunk 1]
```diff
@@ -44,6 +45,13 @@ export class CheckboxControlValueAccessor implements ControlValueAccessor {
     setProperty(this.renderer, this.elementRef, "checked", value);
   }
 
+  get ngClassUntouched() { return isPresent(this.cd.control) ? this.cd.control.untouched : false; }
```
Comment (vicb) [inline modules/angular2/src/forms/directives/checkbox_value_accessor.ts]: `: return Type`

File: modules/angular2/test/change_detection/parser/parser_spec.ts [hunk 2]
```diff
@@ -39,6 +39,12 @@ export function main() {
     return createParser().parseInterpolation(text, location);
   }
 
+  function parseAccessMember(text, location = null): any {
+    return createParser().parseAccessMember(text, location);
+  }
+
+  function unparse(ast): string { return new Unparser().unparse(ast); }
```
Comment (vicb) [inline modules/angular2/test/change_detection/parser/parser_spec.ts]: `ast: AST`

File: modules/angular2/test/change_detection/parser/parser_spec.ts [hunk 3]
```diff
@@ -620,6 +626,22 @@ export function main() {
...
     });
 
+    describe("parseAccessMember", () => {
+      it("should parse a field access", () => {
+        var p = parseAccessMember("name");
+        expect(unparse(p)).toEqual("name");
+      });
+
+      it("should throw when the given expression is an invalid field name", () => {
+        expect(() => parseAccessMember("343")).toThrowError(new RegExp("Unexpected token 343"));
```
Comment (vicb) [inline modules/angular2/test/change_detection/parser/parser_spec.ts]: Should we have the same error message as below ?

File: modules/angular2/test/render/dom/compiler/directive_parser_spec.ts [hunk 4]
```diff
@@ -103,6 +104,11 @@ export function main() {
       expect(ast.source).toEqual('dirProp');
     });
 
+    it('should throw when parsing invalid host properties', () => {
+      expect(() => process(el('<input some-decor-with-invalid-host-props>')))
+          .toThrowError(new RegExp("Invalid access member expression"));
```
Comment (vicb) [inline modules/angular2/test/render/dom/compiler/directive_parser_spec.ts]: single quote pls

**Comment by vicb** [comment]
LGTM with a few comments. The most important one being the exception message.

File: modules/angular2/src/change_detection/parser/parser.ts [hunk 5]
```diff
@@ -487,6 +493,18 @@ class _ParseAST {
     }
   }
 
+  parseAccessMember() {
```
Comment (matanlurey) [inline modules/angular2/src/change_detection/parser/parser.ts]: : AccessMember

File: modules/angular2/src/forms/directives/default_value_accessor.ts [hunk 6]
```diff
@@ -47,6 +48,13 @@ export class DefaultValueAccessor implements ControlValueAccessor {
     setProperty(this.renderer, this.elementRef, 'value', this.value);
   }
```
Comment (matanlurey) [inline modules/angular2/src/forms/directives/default_value_accessor.ts]: Type all of these as :bool

File: modules/angular2/src/forms/directives/select_control_value_accessor.ts [hunk 7]
```diff
@@ -57,6 +58,13 @@ export class SelectControlValueAccessor implements ControlValueAccessor {
     setProperty(this.renderer, this.elementRef, "value", value);
   }
 
+  get ngClassUntouched() { return isPresent(this.cd.control) ? this.cd.control.untouched : false; }
```
Comment (matanlurey) [inline modules/angular2/src/forms/directives/select_control_value_accessor.ts]: Ditto above

File: modules/angular2/src/forms/directives/default_value_accessor.ts [hunk 8]
```diff
@@ -47,6 +48,13 @@ export class DefaultValueAccessor implements ControlValueAccessor {
...
 
+  get ngClassUntouched() { return isPresent(this.cd.control) ? this.cd.control.untouched : false; }
```
Comment (matanlurey) [inline modules/angular2/src/forms/directives/default_value_accessor.ts]: Type all of these as :bool

File: modules/angular2/src/render/dom/compiler/directive_parser.ts [hunk 9]
```diff
@@ -168,8 +168,8 @@ export class DirectiveParser implements CompileStep {
   }
 
   _bindHostProperty(hostPropertyName, expression, compileElement, directiveBinderBuilder) {
```
Comment (matanlurey) [inline modules/angular2/src/render/dom/compiler/directive_parser.ts]: While you're at it, type :void

**Pull request merged**